### PR TITLE
[stable/elasticsearch] Add user/password option for basic auth on client ingress

### PR DESCRIPTION
**Pull request by donbowman**
If `client.ingress.user` and `client.ingress.password` are set,
and client ingress is enabled, HTTP basic authentication will be
enabled on the client ingress.

This allows for (modest) securing of external access to the system.

Signed-off-by: Don Bowman <db@donbowman.ca>

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x ] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x ] Chart Version bumped
- [ x] Variables are documented in the README.md

### CODE REVIEW

**Comment by desaintmartin** [comment]
Thanks!
/assign
/ok-to-test

Could you rename the PR so that it begins with "[stable/elasticsearch]" ?

**Comment by desaintmartin** [comment]
Could you add definition of those new variables in values.yaml, even if commented out?

File: stable/elasticsearch/README.md [hunk 1]
```diff
@@ -92,6 +92,8 @@ The following table lists the configurable parameters of the elasticsearch chart
 | `client.initResources`               | Client initContainer resources requests & limits                    | `{}`                                                |
 | `client.additionalJavaOpts`          | Parameters to be added to `ES_JAVA_OPTS` environment variable for client | `""`                                           |
 | `client.ingress.enabled`             | Enable Client Ingress                                               | `false`                                             |
+| `client.ingress.user`                | If this & password are set, enable basic-auth on ingress            | `nil`                                               |
```
Comment (desaintmartin) [inline stable/elasticsearch/README.md:95]: Actually, this is activated even if password is not set and only user is set. Is that intended?

Comment (donbowman) [reply stable/elasticsearch/README.md:95]: If only the user is set, the password is blank. I guess I could check that case, but then a blank password is disallowed.

So yes, its what i intended, but am not really sure if there's value in it. Do you want me to add an if user and password check to both? I think it works as-is

```
--- /dev/null
+++ b/stable/elasticsearch/templates/client-auth.yaml
@@ -0,0 +1,11 @@
+{{- if .Values.client.ingress.user -}}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: '{{ include "elasticsearch.client.fullname" . }}-auth'
+type: Opaque
+data:
+  auth: {{ printf "%s:{PLAIN}%s\n" .Values.client.ingress.user .Values.client.ingress.password | b64enc | quote }}
+{{- end }}
+
diff --git a/stable/elasticsearch/templates/client-ingress.yaml b/stable/elasticsearch/templates/client-ingress.yaml
index 0564589f..8abc2ce6 100644
--- a/stable/elasticsearch/templates/client-ingress.yaml
+++ b/stable/elasticsearch/templates/client-ingress.yaml
@@ -15,6 +15,11 @@ metadata:
   annotations:
 {{ toYaml . | indent 4 }}
 {{- end }}
+{{- if .Values.client.ingress.user -}}
+    nginx.ingress.kubernetes.io/auth-type: basic
+    nginx.ingress.kubernetes.io/auth-secret: '{{ include "elasticsearch.client.fullname" . }}-auth'
+    nginx.ingress.kubernetes.io/auth-realm: "Authentication-Required"
+{{- end }}
```

**Comment by etiennejournet** [comment]
Why the {PLAIN} directive ? It would be better to hash the password in yml files for version control such as here :  https://github.com/kubernetes-retired/contrib/tree/master/ingress/controllers/nginx/examples/auth

**Pull request merged**